### PR TITLE
docs: Update install for pypi package

### LINKED ISSUE #1208: doc bug: update installation guidance from github to pypi
We now publish `instructlab` to PyPi - we should update all of our docs to _instruct_ users to install from PyPi instead of GitHub.

Followup from #972

### PR DESCRIPTION
**Pull request by hickeyma**
[PyPi package](https://pypi.org/project/instructlab/) is being built since v0.15.1.

This PR updates the instructions to install from package instead of the GitHub repo.

Resolves #1208

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,7 +161,7 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
       python3 -m venv --upgrade-deps venv
       source venv/bin/activate
       (venv) $ pip cache remove llama_cpp_python
-      (venv) $ pip install git+https://github.com/instructlab/instructlab.git@stable -C cmake.args="-DLLAMA_METAL=on"
+      (venv) $ pip install instructlab
```
Comment (leseb) [inline README.md:164]: Do we need the `-C cmake.args="-DLLAMA_METAL=on"` flag or do we build natively with it?

**Review by leseb — COMMENTED** [review]
It'd be great if the commit message and the PR description could share the same content.

**Comment by leseb** [comment]
Also, I believe this resolves https://github.com/instructlab/instructlab/issues/1208, right? It'd be nice to mention it. Thanks!

File: README.md (hunk 1 above)
Comment (hickeyma) [reply README.md:164]: AFIK, no. It enabled by default.

**Comment by hickeyma** [comment]
> Also, I believe this resolves https://github.com/instructlab/instructlab/issues/1208, right? It'd be nice to mention it. Thanks!

Thanks for the heads up @leseb. Added context to the description.

File: README.md (hunk 1 above)
Comment (bjhargrave) [reply README.md:164]: The cmake args are for llama_cpp_python which is a dependency of instructlab. So I think it is important to retain the cmake args here.

Comment (bjhargrave) [reply README.md:164]: Better cmake args are 

```
-C cmake.args="-DLLAMA_METAL=on;-DCMAKE_OSX_ARCHITECTURES=arm64;-DCMAKE_APPLE_SILICON_PROCESSOR=arm64"
```

which handles some odd macOS installs which were migrations from intel macs to arm macs which are some how configured to target x86_64. See https://github.com/instructlab/instructlab/discussions/1148

**Comment by nathan-weinberg** [comment]
@hickeyma Thanks for picking this up - might we want to leave any detail for users who may want to install from `main` or do local development and install directly from the local requirements file?

**Comment by hickeyma** [comment]
> @hickeyma Thanks for picking this up - might we want to leave any detail for users who may want to install from main or do local development and install directly from the local requirements file?

@nathan-weinberg This is covered in the [contributing guide](https://github.com/instructlab/instructlab/blob/main/CONTRIBUTING/CONTRIBUTING.md#development). It was removed from the readme once the contributing guide was added.

File: README.md (hunk 1 above)
Comment (bjhargrave) [reply README.md:164]: The recent update of llama_cpp_python to 0.2.75 includes updates to llama.cpp which now defaults to "LLAMA_METAL=on". So we don't need this anymore.

**Review by bjhargrave — APPROVED** [review]

**Review by kelbrown20 — COMMENTED** [review]
From a docs perspective, this LGTM. I think someone on the technical side should take a look as well

**Review by leseb — COMMENTED** [review]
I still would like to see this comment being addressed https://github.com/instructlab/instructlab/pull/1214#pullrequestreview-2082582412, then LGTM :), thanks in advance!

**Comment by hickeyma** [comment]
> I still would like to see this comment being addressed https://github.com/instructlab/instructlab/pull/1214#pullrequestreview-2082582412, then LGTM :), thanks in advance!

@leseb What needs to be addressed?

**Comment by leseb** [comment]
> > I still would like to see this comment being addressed [#1214 (review)](https://github.com/instructlab/instructlab/pull/1214#pullrequestreview-2082582412), then LGTM :), thanks in advance!
> 
> @leseb What needs to be addressed?

Just amend your commit to match the body of the PR. You've put useful information that needs to be retained in git. That:

```
[PyPi package](https://pypi.org/project/instructlab/) is being built since v0.15.1.

This PR updates the instructions to install from package instead of the GitHub repo.

Resolves https://github.com/instructlab/instructlab/issues/1208
```

Thanks!

**Review by markstur — APPROVED** [review]

**Comment by hickeyma** [comment]
> Just amend your commit to match the body of the PR. You've put useful information that needs to be retained in git.

Done

**Review by leseb — APPROVED** [review]

**Comment by leseb** [comment]
> > Just amend your commit to match the body of the PR. You've put useful information that needs to be retained in git.
> 
> Done

Thanks!

**Pull request merged**